### ISSUE TITLE
Support for replacing field's struct type with `optfield`-variant

### ISSUE THREAD

**Issue by Holzhaus** [roignpar/optfield#11]
Hi,

I'm trying to use `optfield` for serde config parsing, so that I can parse the default config from the non-optional struct variant, and the override default options from the user config which might lack some values for which the defaults are sufficient.

I would be nice if `optfield` could replace field types with their `optfield`-variant.

Let's say I have a config struct like this:

```rust
struct Config {
    server_url: String,
    window: WindowConfig,
}

struct WindowConfig {
    width: usize;
    height: usize
}
```

Using `optfield`, I can get this:

```rust
struct ConfigOpt {
    server_url: Option<String>,
    window: Option<WindowConfig>
}

struct WindowConfigOpt {
    width: Option<usize>,
    height: Option<usize>,
}
```

But what I actually want:

```rust
struct ConfigOpt {
    server_url: Option<String>,
    window: Option<WindowConfigOpt>
}

struct WindowConfigOpt {
    width: Option<usize>,
    height: Option<usize>,
}
````

Would it be feasible to add functionality like this?

**Comment by roignpar** [comment]
Hi,

Unfortunately this is not supported by optfield and probably never will be.
There has been an attempt to add such functionality, but it didn't work out: https://github.com/roignpar/optfield/pull/2.
TL;DR: use https://github.com/lesurp/OptionalStruct instead.
